adds gulp, babel, browserify

SylvanSign opened the pull request:
`gulp` is a build tool that lets you automate certain repetitive workflows. In this case, it helps automate the standard workflow of
```
   make changes -> save files -> babelify -> browserify -> restart server
```
It could probably use some tweaking, like only restarting the server if our backend files have changed & only building when our frontend files have changed. I can work on that later.
**edit:** I've got it restarting server on backend changes and rebuilding frontend scripts when they change now

For now though, you just run `npm run watch` (or `gulp` if you install gulp globally) and it'll watch for any code change. When it sees a change, it'll transpile our code from ES6 -> ES5, bundle our frontend code and all it's `require`'d dependencies into one file (`public/bundle.js`), which is the only file we'll need to include in our html.

I can go over it with y'all some point this week. But I think this should make our lives a lot easier. Shouldn't be horrible to add in extra stuff if necessary (though I couldn't figure out how to automatically beautify without infinite looping...)

SylvanSign committed:
adds gulp, babel, browserify

SylvanSign committed:
defer script loading

ZakMiller reviewed (changes requested):
Looks like there is a conflict?

SylvanSign commented:
Yeah, I removed the nodemon stuff in favor of gulp-nodemon so that's probably the issue.I''ll fix it after work.

SylvanSign commented:
fixed the conflict and rebased.

How do you all feel about fbb3799? I sort of like separating the event handling from the html tags entirely. So instead of `<button onClick='myFunction()'>` we do `<button id="b1">` and add a `click` listener for that tag in our script files later. It feels like better compartmentalization to me, but I don't know what the convention is.

SylvanSign commented on gulpfile.js:
```diff
@@ -0,0 +1,69 @@
+    .pipe(sourcemaps.init({
+      loadMaps: true
+    }))
+    .pipe(sourcemaps.write('./'))
```
sourcemaps are used by browsers to map errors back to the original, unbundled/unminified, scripts. So it won't tell you to look at line `900` of our giant `bundle.js`, it'll tell you to look at line `13` of `foo.js`. Pretty nifty

ZakMiller replied:
Cool
